### DISCUSSION TITLE
Firebase AppCheck Debug Token Changes Upon App Reinstallation

### DISCUSSION THREAD

**Discussion by queendevelopers** [firebase/flutterfire#12336] [Q&A]
I'm facing an issue where the Firebase AppCheck debug token changes whenever the app is reinstalled. We utilize the DebugProvider for our development and QA builds. However, this frequent token change poses a challenge as we constantly need to update the debug token in the Firebase console, which is not reliable. How can we effectively mitigate this issue? Thanks!

**Comment by orbachar** [comment]
did you find a solution? having the same issue

**Reply by queendevelopers** [reply]
Yes, for temporary solution, I used firebase REST API and it solved the issue. Probably this might not be the good solution but workaround for now.

**Comment by cnkygmr** [comment]
I'm having the same problem, it keeps renewing, I add it, it doesn't work directly when added, and the token changes when I recompile. In the past, it remained the same until the emulator was closed and opened again.

**Comment by cnkygmr** [comment]
flutter clean solves the problem
